### PR TITLE
Eason/ticket 102: backend testing

### PR DESCRIPTION
**Pull request by WYC401**
## Description

This pr create unit tests for backend testing.

### Changes

* add a folder `./test/unit/controllers` to test all controllers
* add some `await` keyword in controllers

### Updates

under review

## Checklist

- [x] Ran `npm run lint:fix` to format code
- [x] Assigned PR to all authors
- [x] Requested at least one reviewer

### CODE REVIEW

**Comment by WYC401** [comment]
[backend testing](https://trello.com/c/PI97ND6T/102-backend-testing)

File: controllers/projectController.js [hunk 1]
```diff
@@ -1,158 +1,159 @@
...
-    }
-  } catch (error) {
-    res.status(500).json({
-      message: "Internal server error while attempting to create project",
-      errCode: PROJECT_ERR.PROJECT001,
-      error,
-    });
-  }
+			//TODO: should add await
+			await newProject
```
Comment (ElisaHu1) [inline controllers/projectController.js:22]: why adding await here? some comments may be needed

**Review by ElisaHu1 — COMMENTED** [review]
fabulous!

**Review by ElisaHu1 — APPROVED** [review]
approve for me, wait for @svolterra

**Review by svolterra — COMMENTED** [review]
Just pausing the code review due to the following concerns:
1. When I'm on this branch I begin getting errors in my controller files (as seen in the screenshot below)
2. One of the test suites fails for me 😔 

local branch errors

Does this happen on your end as well? Additionally, do you know why it may be happening? 🤔

**Comment by ElisaHu1** [comment]
@WYC401 any update for this pr?

**Comment by WYC401** [comment]
> @WYC401 any update for this pr?

I was trying to make changes only on unit tests without adding `await`  keywords to controllers. But it seems no way. with `await`, the api response time doesn't increase so let's merge this if this looks good to you

what do you  think? @ElisaHu1 @svolterra

**Comment by ElisaHu1** [comment]
image

**Comment by WYC401** [comment]
> image

I promise this time it won't fail again. It is caused by that someone created the same datapoint in the database so that it caused name collision. I change the snapshot title name to 'npmTest' so that no one will make up the same object as me lol.

**Comment by ElisaHu1** [comment]
eason can you fix the conflict and update the member test accordingly?

**Pull request merged**